### ISSUE TITLE
Italic variant out of a custom version of Fira Mono

### ISSUE THREAD

**Issue by brainplot** [tonsky/FiraCode#513]
I know that this isn't from the original Fira Mono repository but this guy did a rather good job making a custom italic variant of Fira Mono.

https://github.com/zwaldowski/Fira/tree/zwaldowski/mod-new/otf

I've downloaded Fira Mono Italic and tweaked its name to "Fira Code" via FontForge so that it appears as the real italic variant of Fira Code in my font manager.
By doing this, my text editors use the custom version every time they need to use the italic variant (such as in comments or static methods/fields). This is the result:

![fira-italic](https://user-images.githubusercontent.com/28014843/31862013-340ce796-b737-11e7-8c99-98df9df92164.png)

To me it looks way better than all the editors trying to fake it. The problem is that there are no ligatures (as shown in the picture) because it's Fira Mono, obviously.
What do you think about creating Fira Code Italic out of that?
The original Fira Mono Italic will probably come out, eventually, but in the meanwhile this might be an acceptable alternative.

**Comment by sohnryang** [comment]
Well, I think it'd be good if @tonsky adds ligatures on the custom Fira Mono Italic font.

**Comment by brainplot** [comment]
> Well, I think it'd be good if @tonsky adds ligatures on the custom Fira Mono Italic font.

Yeah, I stopped using this font because of this basically: the italic variant was just too ugly to look at.
Unfortunately, looks like this is not on his roadmap since this issue has been open for quite some time without receiving any replies; but that's fine. Mine was nothing more than a suggestion.

**Comment by tonsky** [comment]
I’m not against it, but it’s a lot of work. Basically re-creating each ligature in italics. Maybe one day...

**Comment by sohnryang** [comment]
Huh? [carbon.now.sh](https://carbon.now.sh) supports Italic variant of Fira Code.

**Comment by sohnryang** [comment]
![carbon](https://user-images.githubusercontent.com/15506585/50596367-65d1e700-0ee7-11e9-8c81-07c0116ac2fd.png)

**Comment by j-f1** [comment]
That’s just a skewed version of the font.

**Comment by brainplot** [comment]
I second what @j-f1 says.

@sohnryang, if the word `log` was italic, in your example, you would see how bad it looks. It becomes immediately evident that the font doesn't have its own italic variant and the editor tries its best to make it look like it does (and the result is not consistent across all editors, probably that's one reason why it looks really bad). It's even more evident when you compare it with a font that does have an italic variant.

**Comment by Avi-D-coder** [comment]
I made a corrected auto italic version [FiraCode-italic](https://github.com/Avi-D-coder/FiraCode-italic). There are a few things left to fix, but it is very usable.

**Comment by j-f1** [comment]
@Avi-D-coder Awesome! My one suggestion would be to take inspiration from Fira Sans for some of the lowercase letters, which have different forms in italic:

<img width="587" alt="Fira Sans regular and italic lowercase letters" src="https://user-images.githubusercontent.com/25517624/65701500-61d68e00-e04f-11e9-80f3-0f66782a0830.png">
<img width="587" alt="Same as above, but with differences highlighted" src="https://user-images.githubusercontent.com/25517624/65701734-ce518d00-e04f-11e9-927f-d559d5a372d1.png">

All the other letters are identical for roman/italic forms in Fira Sans.

**Comment by Avi-D-coder** [comment]
@j-f1 I'll look into fixing it, soonish.

**Comment by Avi-D-coder** [comment]
There are more character differences, I am going to scale all the sans italics to make them mono spaced rather than patching the italicized mono. It will look better, but its going to take some time.

**Comment by Avi-D-coder** [comment]
@j-f1 Take a look, I changed the requested characters

**Comment by j-f1** [comment]
I love it! A few things:

- the line height seems to be different from the regular Fira Code (in this case, the variable font version)
- the `r` is missing its serif
- stylistic sets (i.e. for `$`) don’t seem to be supported

Did you do the initial conversion from Fira Code v2 or an older version?

<img width="667" alt="Screen Shot 2019-09-28 at 15 02 03" src="https://user-images.githubusercontent.com/25517624/65821238-f3393200-e200-11e9-8c15-e4c59b20c1dc.png">

**Comment by Avi-D-coder** [comment]
@j-f1 I presently don't support style sheets. To quote the readme "This is my manually corrected and auto Ligaturized, version of @zwaldowski's synthesized italic version of Mozilla's Fira Mono."

If you want to add style sheets, I would gladly except a PR. The font source is in [Avi-D-coder/fira-mono-italic](https://github.com/Avi-D-coder/fira-mono-italic). Once there is a mono-italic version with style sheets, you or I could run it through [Ligaturizer](https://github.com/Avi-D-coder/Ligaturizer).

**Comment by scott-ln** [comment]
@Avi-D-coder Thanks a lot for this! However I just installed it on Windows and text that was previously italicized in my editor (Sublime Text) has now stopped being in italics. Do you know if there's something else that I need to do?

**Comment by Avi-D-coder** [comment]
@scott-ln I don't have Windows so it has not been tested with it. In Linux installing the files from the repo is enough

**Comment by scott-ln** [comment]
No worries, thank you. I wonder if @tonsky may know why having compiled a support table for Fira Code in various editors.

**Comment by tonsky** [comment]
I have no idea. Wrong name for italics variant?

**Comment by Avi-D-coder** [comment]
@scott-ln could you screen shot your font manager

**Comment by scott-ln** [comment]
The italic variants are picked up in Windows settings:

![Fira Code in settings](https://user-images.githubusercontent.com/43204121/66400039-a3b0ee00-e9d8-11e9-9421-2553ed9fdd6d.png)

However I can see that if you open the font files by double-clicking to preview them (the pre-Windows 10 way), "Font name" appears only as "Fira Code", so that may well be the issue. Opening other members of the family this way shows "Fira Code Light", etc.

![Fira Code RegularItalic open](https://user-images.githubusercontent.com/43204121/66400279-18842800-e9d9-11e9-86f4-823146a77050.png)

![Fira Code BoldItalic open](https://user-images.githubusercontent.com/43204121/66400111-c6db9d80-e9d8-11e9-8395-00980936ee92.png)

**Comment by Avi-D-coder** [comment]
Are you using the most recent version from my repo?

**Comment by scott-ln** [comment]
Yes, from [`fira-mono-italic/distr/otf/`](https://github.com/Avi-D-coder/fira-mono-italic/tree/master/distr/otf).

**Comment by scott-ln** [comment]
Very happy to report that after a system update that involved a restart, the issue has gone away. Just Windows being Windows, I guess. Thanks again for your work Avi!

**Comment by scott-ln** [comment]
Hi, sorry, me again. Now that the italic characters are visible, it appears that their baseline is lower than the other variants.

![image](https://user-images.githubusercontent.com/43204121/66572471-eefa0680-eb68-11e9-8126-f4aa690c6791.png)

**Comment by tomasdev** [comment]
Are there plans to make italic become part of FiraCode officially?

**Comment by tonsky** [comment]
> Are there plans to make italic become part of FiraCode officially?

Not unless they are added to Fira Mono

**Comment by brainplot** [comment]
> Not unless they are added to Fira Mono

Sadly I don't see this happening, ever. I've looked at the [repo](https://github.com/mozilla/Fira) and it doesn't seem to be actively developed.

**Comment by TBBle** [comment]
FiraGO still has the entirety of 'Mono' as a TODO, and nothing's shifted in that respect since 2018, when they said explicitly that they weren't funded for that in https://github.com/bBoxType/FiraGO/issues/16#issuecomment-400938129
